errai-security/errai-security-keycloak/src/main/java/org/jboss/errai/security/keycloak/KeycloakAuthenticationService.java

```diff
@@ -230,7 +230,17 @@ protected User createKeycloakUser(final AccessToken accessToken) {
   }
 
   private Collection<? extends Role> createRoles(final AccessToken accessToken) {
-    Set<String> roleNames = accessToken.getResourceAccess(accessToken.getIssuedFor()).getRoles();
+    Set<String roleNames = new HashSet<>();
+    //Add app roles first, if any
+    AccessToken.Access access = accessToken.getResourceAccess(accessToken.getIssuedFor());
+    if(access!=null && access.getRoles()!=null){
```

Comment:
To be consistent with the project's code formatting, this line should be formatted like so:
`if (access != null && access.getRoles() != null) {`